### PR TITLE
fix: convert_module_to_trt_engine

### LINKED ISSUE #2692: ✨[Bug] Add lowering, decompositions, nn.Module support for convert_module_to_trt_engine API
**Is your feature request related to a problem? Please describe.**
Add lowering, decompositions support, nn.Module support for `convert_module_to_trt_engine` API. Currently the  `convert_module_to_trt_engine` API assumes the input is a graph_module

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by zewenli98**
# Description

fix: `convert_module_to_trt_engine`
add doc.

Fixes #2692

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [x] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/_compile.py [hunk 1]
```diff
@@ -319,10 +319,21 @@ def convert_method_to_trt_engine(
             "convert_method_to_trt_engine call is not supported for ir=fx"
         )
     elif target_ir == _IRType.dynamo:
+        # Prepare torch and torchtrt inputs
+        import collections.abc
```
Comment (narendasan) [inline py/torch_tensorrt/_compile.py:323]: Import this at top
